### ISSUE TITLE
gcForest 1.1 is here

### ISSUE THREAD

**Issue by kingfengji** [kingfengji/gcForest#23]
Hi all.
I've updated the package. Previously I was working on a project about extracting information along decision paths. The paper just got accepted in AAAI-18(called encoderForest). Tonight I finally got time to update the code.
Please give me feed backs if you are in trouble.

**Comment by zhaopanpan73** [comment]
so excited! thanks your effort!

**Comment by SecMatrix** [comment]
Hi, FengJi. Express my gratitude for your efforts first. 

Would you like to share code of the encoderForest in github? I'm very interested in the new forest structure. 🙂

Moreover, could you give a suggestion about the basic hardware requirements for gcForest? such as the minimum memory needs. I find it's hard for my PC to run the code.

**Comment by kingfengji** [comment]
@SecMatrix thanks .
Glad you ask about eForest, it'll be pretty fast, any laptop can run without trouble.
for the gcForest, the main bottleneck for RAM is the fine-grained stage, I would recommend a 32/64G desktop, if your laptop is below this storage, set the keep in mem to FALSE and it will work fine.
If your task only need cascading structure, 8/16 G will be OK.( or use xgboost as base estimators, it is much small in terms of mem issue.)

**Comment by pylablanche** [comment]
Hi @kingfengji ,

Really nice work!!!
Have you tried running it with python 3.5 ?

**Comment by kingfengji** [comment]
Thanks!
I think py3.5 will be working just fine, if I remember it correctly... in
other words, I think at most one or two easy fix ups.( possibly zero
changes)

2017-12-30 18:54 GMT+08:00 Pierre-Yves Lablanche <notifications@github.com>:

**Comment by SecMatrix** [comment]
Hi @kingfengji ,

I have a childish question，
Why did you remove completely-random tree forests structure in this version? I know little about this forest.

**Comment by pylablanche** [comment]
@kingfengji ,

I have downloaded everything and will try using it with python 3.6. Will let you know if I face any problem.

**Comment by kingfengji** [comment]
Hi, I did not.

for completely random tree forest, just use:

ExtraTreesClassifier and set max_features = 1.

Thanks

2018-01-03 9:33 GMT+08:00 Zihoo <notifications@github.com>:

**Comment by kingfengji** [comment]
This is great, I'm looking forward of your feedbacks!!

Thanks!!

2018-01-06 0:51 GMT+08:00 Pierre-Yves Lablanche <notifications@github.com>:

**Comment by kingfengji** [comment]
@SecMatrix Hi, I did not.

for completely random tree forest, just use:

ExtraTreesClassifier and set max_features = 1.

Thanks

**Comment by pylablanche** [comment]
@kingfengji ,
I have tried to run the demo_mnist and the code cannot work under python3 because of some modules that are no longer available with python3. You can therefore officially announce that the code is not compatible with python 3.

Nonetheless correcting these bugs will be pretty straightforward (hopefully).
I will let you know when I'm done.

**Comment by SecMatrix** [comment]
LOL
Thank you very much.

**Comment by zhuyf8899** [comment]
Hi Mr. @kingfengji , firstly I want to thank you for your great Innovation.
As a freshman in this domain, I have a question about expansibility: Can this model be extend to a regression model? if it has the possibility, how should I modify it?

Thanks again for your attention.

**Comment by pylablanche** [comment]
Hi @kingfengji ,

I face a little problem when trying to load the "xgboost" library in lib/gcforest/estimators/sklearn_estimators.py (basically it is missing)
I just would like to know if it the same package as the following one ?
xgboost (0.7.post3)           - XGBoost Python Package

If so maybe you should include it in the requirements.txt :)

**Comment by kingfengji** [comment]
@pylablanche You are absolutely right as always, I'll add it .
I've tested python 3.5 and it seems all right.

**Comment by pylablanche** [comment]
@kingfengji ,
I was at first a bit surprised to read that you can run your code with python3.5 as it used to include functions specific to python2.7 (such as "xrange" and "basestring") but I see that you've edited your comment so I'm going to give it a new try... as soon as I have fixed my xgboost installation :/

Looking forward to giving you more feedback (if necessary)

**Comment by kingfengji** [comment]
@pylablanche 
yes, I modified such places to make it compatible for python3.5,
i think xgboost can be installed via pip, as long as the gcc version is the right one, otherwise it is indeed a time consuming procedure.....

**Comment by pylablanche** [comment]
@kingfengji 
I'm indeed facing problems to install xgboost (pip fails so far) that are absolutely not related to gcForest. I'll fix it and test your package soon!

**Comment by irfan01000** [comment]
i want to work on your project for my phd..need your suggestion and future enhancement
